git_machete/tests/functional/test_machete.py

```diff
@@ -47,122 +47,147 @@ def new_request(self) -> "MockGithubAPIRequest":
         return MockGithubAPIRequest(self)
 
     def get_issue(self, issue_no: str) -> Optional[Dict[str, Any]]:
-        for index, issue in enumerate(self.issues):
+        for issue in self.issues:
             if issue['number'] == issue_no:
-                return self.issues.pop(index)
+                return issue
         return None
 
     def get_pull(self, pull_no: str) -> Optional[Dict[str, Any]]:
-        for index, pull in enumerate(self.pulls):
+        for pull in self.pulls:
             if pull['number'] == pull_no:
-                return self.pulls.pop(index)
+                return pull
         return None
 
 
-class ResponseWrapper:
-    def __init__(self, data: Union[List[Dict[str, Any]], Dict[str, Any]]) -> None:
-        self.data: Union[List[Dict[str, Any]], Dict[str, Any]] = data
+class MockGithubAPIResponse:
+    def __init__(self, status_code: int, response_data: Union[List[Dict[str, Any]], Dict[str, Any]]) -> None:
+        self.response_data: Union[List[Dict[str, Any]], Dict[str, Any]] = response_data
+        self.status_code: int = status_code
 
-    def read(self) -> Union[str, bytes]:
-        return json.dumps(self.data).encode()
+    def read(self) -> bytes:
+        return json.dumps(self.response_data).encode()
 
 
 class MockGithubAPIRequest:
+    def __init__(self, github_api_state: MockGithubAPIState) -> None:
+        self.github_api_state: MockGithubAPIState = github_api_state
 
-    def __init__(self, git_api_state: MockGithubAPIState) -> None:
-        self.git_api_state: MockGithubAPIState = git_api_state
-
-    def __call__(self, url: str, headers: Dict[str, str] = None, data: Union[str, bytes, None] = None, method: str = '') -> "ResponseWrapper":
+    def __call__(self, url: str, headers: Dict[str, str] = None, data: Union[str, bytes, None] = None, method: str = '') -> "MockGithubAPIResponse":
         self.parsed_url: ParseResult = urlparse(url, allow_fragments=True)
         self.parsed_query: Dict[str, List[str]] = parse_qs(self.parsed_url.query)
-        self.status_code: Optional[int] = None
         self.json_data: Union[str, bytes] = data
         self.return_data: Optional[Union[List[Dict[str, Any]], Dict[str, Any]]] = None
         self.headers: Dict[str, str] = headers
-        self.handle_method(method)
-        return ResponseWrapper(self.return_data)
+        return self.handle_method(method)
 
-    def handle_method(self, method: str) -> None:
+    def handle_method(self, method: str) -> "MockGithubAPIResponse":
         if method == "GET":
-            self.handle_get()
+            return self.handle_get()
         elif method == "PATCH":
-            self.handle_patch()
+            return self.handle_patch()
         elif method == "POST":
-            self.handle_post()
+            return self.handle_post()
+        else:
+            return self.make_reponse_object(HTTPStatus.NOT_FOUND, [])
```

Comment:
`response`

Reply:
Also, nit: real Github is likely to return 405 Method Not Allowed status in this case